main/autogen/autogen.h

```diff
@@ -81,13 +85,22 @@ struct ParsedFile {
 
     std::string toString(core::Context ctx);
     std::string toMsgpack(core::Context ctx, int version);
-    void classlist(core::Context ctx, std::vector<std::string> &out);
+    std::vector<std::string> listAllClasses(const core::Context ctx);
+    std::optional<AutogenSubclassMap> listAllSubclasses(const core::Context ctx,
+                                                        const std::vector<std::string> &absolutePathsToIgnore,
+                                                        const std::vector<std::string> &relativePathsToIgnore);
 
 private:
     std::vector<core::NameRef> showFullName(core::Context ctx, DefinitionRef id);
     friend class MsgpackWriter;
 };
 
+void maybeInsertChild(const std::string &parentName, const AutogenSubclassSet &children, AutogenSubclassMap &out);
```

Comment:
could those methods be static members of some class? We try to not define free-floating public functions

Reply:
Done in https://github.com/sorbet/sorbet/pull/1040/commits/bc902b3e9d7ef2ed54b581f99a5a49018f3d3485